tests/utils.py

```diff
@@ -465,3 +466,10 @@ def assert_rpc_response(response, with_key="result"):
     assert response and 200 == response.status_code
 
     assert with_key in response.json.keys(), str(response.json)
+
+
+def assert_valid_cwl(cwl: Path) -> str:
+    """Run cwltool --validate --strict on a cwl file and checks if the file is valid."""
+    output = str(check_output(["cwltool", "--validate", "--strict", str(cwl.resolve())]))
```

Comment:
cwltool is written in Python, I think you can just use 
```
from cwltool.context import LoadingContext
from cwltool.workflow import default_make_tool
from cwltool.resolver import tool_resolver
from cwltool.load_tool import load_tool

context = LoadingContext({"construct_tool_object": default_make_tool,
                               "resolver": tool_resolver})
parsed = load_tool(path, context)
```

and that will raise an exception

Reply:
Ok great. I tried to check how I can do this but gave up too easily. Thanks for figuring it out.